.ci/pitest.groovy

```diff
@@ -0,0 +1,302 @@
+            exitCode = 1
+            println "Surviving mutation/s found:"
+            survivedAndUnsuppressedMutations.each { println it }
+        }
+        Set<Mutation> killedAndSuppressedMutations =
+                differenceBetweenSets(suppressedMutations, survivedMutations)
+        if (!killedAndSuppressedMutations.isEmpty()) {
+            exitCode = 1
+            println "\nUnnecessary suppressed mutations/s found:"
+            killedAndSuppressedMutations.each { println it }
+        }
+    }
+    return exitCode
+}
+
+private static Set<Mutation> differenceBetweenSets(final Set<Mutation> setOne,
+                                                   final Set<Mutation> setTwo) {
+    Set<Mutation> result = new HashSet<Mutation>(setOne)
+    result.removeIf(mutation -> setTwo.contains(mutation))
+    return result
+}
+
+record Mutation(String sourceFile, String mutatedClass,
+                String mutatedMethod, String mutator,
+                String description, String lineContent, int lineNumber)
+        implements Comparable<Mutation> {
+
+    private static final int LINE_NUMBER_NOT_PRESENT_VALUE = -1
+
+    boolean equals(o) {
+        if (this.is(o)) {
+            return true
+        }
+        if (!(o in Mutation)) {
+            return false
+        }
+
+        Mutation mutation = (Mutation) o
+
+        if (lineNumber() != LINE_NUMBER_NOT_PRESENT_VALUE
+                && mutation.lineNumber() != LINE_NUMBER_NOT_PRESENT_VALUE) {
+            if (lineNumber() != mutation.lineNumber()) {
+                return false
+            }
+        }
+        if (description() != mutation.description()) {
+            return false
+        }
+        if (lineContent() != mutation.lineContent()) {
+            return false
+        }
+        if (mutatedClass() != mutation.mutatedClass()) {
+            return false
+        }
+        if (mutatedMethod() != mutation.mutatedMethod()) {
+            return false
+        }
+        if (mutator() != mutation.mutator()) {
+            return false
+        }
+
+        return sourceFile() == mutation.sourceFile()
+    }
+
+    int hashCode() {
+        int result
+        result = sourceFile.hashCode()
+        int hashCodeMultiplier = 31
+        result = hashCodeMultiplier * result + mutatedClass.hashCode()
+        result = hashCodeMultiplier * result + mutatedMethod.hashCode()
+        result = hashCodeMultiplier * result + mutator.hashCode()
+        result = hashCodeMultiplier * result + description.hashCode()
+        result = hashCodeMultiplier * result + lineContent.hashCode()
+        result = hashCodeMultiplier * result + lineNumber
+        return result
+    }
+
+    @Override
+    String toString() {
+        String toString = """
+Source File: "${sourceFile()}"
+Class: "${mutatedClass()}"
+Method: "${mutatedMethod()}"
+Line Contents: "${lineContent()}"
+Mutator: "${mutator()}"
+Description: "${description()}"
+"""
```

Comment:
I am starting to think that we may be better served by having this just output XML that can be copy-pasted into the suppression file (with appropriate indentation and all). @Vyom-Yadav what do you think?

Reply:
Normal text is easier to read than XML, what about a flag like `--generate-suppression`, though reading XML directly is not a tough job, we can reduce the complexity of adding flags by just outputting XML directly.

Reply:
Done, added the `-g | --generate-suppression flag`